packages/react/lib/internal/useCallSolution.ts

```diff
@@ -0,0 +1,28 @@
+import {
+  Address,
+  ChainID, ContractCallSolutionOptions,
+  type SolutionResponse,
+  Sprinter,
+  TokenSymbol
+} from "@chainsafe/sprinter-sdk";
+import {useCallback} from "react";
+import { useAsyncRequest} from "./useAsyncRequest.ts";
+
+export function useCallSolution(sprinter: Sprinter) {
+  const { state: callSolution, makeRequest } = useAsyncRequest<SolutionResponse>();
+
+  const getCallSolution = useCallback((account: Address,
+                                   destinationChain: ChainID,
+                                   token: TokenSymbol,
+                                   amount: number,
+                                   contractCall: ContractCallSolutionOptions,
+                                   threshold?: number,
+                                   whitelistedSourceChains?: ChainID[],
+  ) => {
```

Comment:
nit: i think this would be more in the spirit of js (+ linter should take care of this)
```suggestion
  const getCallSolution = useCallback((
    account: Address,
    destinationChain: ChainID,
    token: TokenSymbol,
    amount: number,
    contractCall: ContractCallSolutionOptions,
    threshold?: number,
    whitelistedSourceChains?: ChainID[],
  ) => {
```

Reply:
also, why doesn't this method just have one param of type `ContractSolutionOptions` and pass it to SDK?

Reply:
it is shorter to implement

```ts
getCallSolution("0x0", 100, "usdc", 10000, "0x0"); 
```

/w object

```ts
getCallSolution({ account: "0x0", destinationChain: 100, token: "usdc", amount: 10000, contractCall: "0x0"}); 
```

Reply:
but a bit more readable/explicit? 🤷